src/tlo/methods/healthsystem.py

```diff
@@ -193,6 +194,27 @@ class HealthSystem(Module):
             "Availability of beds. If 'default' then use the availability specified in the ResourceFile; if "
             "'none', then let no beds be  ever be available; if 'all', then all beds are always available. NB. This "
             "parameter is over-ridden if an argument is provided to the module initialiser."),
+        'EquipmentCatalogue': Parameter(
+            Types.DATA_FRAME, "Data on equipment items and packages."),
+        'equipment_availability_estimates': Parameter(
+            Types.DATA_FRAME, "Data on the availability of equipment items and packages."
+        ),
```

Comment:
Just another comment from me - having these two DataFrames as parameters means we're actually saving them twice (since we pass the DataFrames into the Equipment constructor, which then takes it's own copy of the DF).

I've opened https://github.com/UCL/TLOmodel/pull/1343 which removes these two parameters and instead just gets the Equipment class to read from the files directly. It also removes a few other methods from the HealthSystem class that aren't needed with this change.

Though if having these dataframes as parameters is required by the logging or output, feel free to ignore this suggestion!

Reply:
> Just another comment from me - having these two DataFrames as parameters means we're actually saving them twice (since we pass the DataFrames into the Equipment constructor, which then takes it's own copy of the DF).

Is this definitely the case? As far as I can see the call to the `Equipment` initialiser just passes references to the parameter dictionary entries and the `Equipment` initialiser itself doesn't directly make any copies, it just attaches the passed references as attributes of the class? I think having these values as parameters of the `HealthSystem` module is consistent with the pattern generally used which is that any values read in from external resource files are parameters (and could feasibly be swapped for other values by changing the resource files or overriding the parameter values in a scenario (though currently the latter case isn't possibly because of the constraint to scalar valued parameters - see #988).

Reply:
Yes, I was thinking the same as @matt-graham  here -- that we want all the reading to happen in `read_parameters` so that the `Scenario` class can overwrite anything afterwards.